cmd/kube-rbac-proxy/app/options/proxyoptions.go

```diff
@@ -125,13 +140,87 @@ func (o *ProxyOptions) ApplyTo(c *server.KubeRBACProxyInfo, a *serverconfig.Auth
 		}
 	}
 
+	c.UpstreamHeaders = o.UpstreamHeader
 	c.IgnorePaths = o.IgnorePaths
 	c.AllowPaths = o.AllowPaths
 	a.APIAudiences = o.TokenAudiences
 
 	return nil
 }
 
+func validateSecureConnection(o *ProxyOptions) error {
```

Comment:
this function does not validate connections

Reply:
I assume you want me to change the name.

I am renaming it to `validateSecureConnectionConfig`.